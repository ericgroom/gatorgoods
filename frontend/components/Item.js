import styled from "styled-components";
import PropTypes from "prop-types";
import { formatPrice } from "../utils";

const ItemWrapper = styled.div`
  border: 2px solid lightgray;
  box-shadow: 0 0 3px lightgray;
  border-radius: 0.5rem;
  padding: 1rem;
  margin: 1rem;

  /* on small screens, show elements in a single column */
  display: grid;
  grid-template-areas:
    "title"
    "image"
    "right";
  text-align: center;
  justify-content: center;

  /* on larger screens, show items with image on the left */
  @media (min-width: 600px) {
    text-align: left;
    grid-template-columns: minmax(100px, 400px) 1fr;
    grid-template-rows: auto 1fr;
    grid-template-areas:
      "image title"
      "image right";
  }

  .title {
    grid-area: title;
  }

  .right {
    display: flex;
    flex-direction: column;
    width: 100%;
    grid-area: right;
  }

  .image-wrapper {
    /* width: 100px; */
    margin: auto 0.5rem;
    flex: initial;
    grid-area: image;
  }

  .price {
    color: green;
    font-weight: 600;
    padding-left: 0.5rem;
  }

  .horizontal {
    width: 100%;
    display: inline-flex;
    justify-content: space-between;
    align-items: center;
    margin-top: auto;
    p {
      margin: 0;
    }
  }
`;

const Item = props => (
  <ItemWrapper>
    <h3 className="title">{props.item.title}</h3>
    <div className="image-wrapper">
      <img
        src={props.item.image}
        alt={props.item.title}
        className="img-fluid"
      />
    </div>
    <div className="right">
      <small>category: {props.item.category}</small>
      <p>{props.item.description}</p>
      <div className="horizontal">
        <button className="btn btn-outline-primary">Contact Seller</button>
        <p className="price">{formatPrice(props.item.price)}</p>
      </div>
    </div>
  </ItemWrapper>
);

Item.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    description: PropTypes.description,
    category: PropTypes.number,
    image: PropTypes.string
  }).isRequired
};

export default Item;
