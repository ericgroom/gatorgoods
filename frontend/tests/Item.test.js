import { shallow } from "enzyme";
import Item from "../components/Item";

describe("Item component", () => {
  it("matches snapshot", () => {
    const item = {
      title: "Boots",
      price: 1200,
      category: 1,
      image: "/a.png",
      description: "Some boots"
    };
    const wrapper = shallow(<Item item={item} />);
    expect(wrapper).toMatchSnapshot();
  });
});
