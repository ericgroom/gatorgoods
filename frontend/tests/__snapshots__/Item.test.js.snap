// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    item={
      Object {
        "category": 1,
        "description": "Some boots",
        "image": "/a.png",
        "price": 1200,
        "title": "Boots",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef>
        <h3
          className="title"
        >
          Boots
        </h3>
        <div
          className="image-wrapper"
        >
          <img
            alt="Boots"
            className="img-fluid"
            src="/a.png"
          />
        </div>
        <div
          className="right"
        >
          <small>
            category: 
            1
          </small>
          <p>
            Some boots
          </p>
          <div
            className="horizontal"
          >
            <button
              className="btn btn-outline-primary"
            >
              Contact Seller
            </button>
            <p
              className="price"
            >
              $12.00
            </p>
          </div>
        </div>
      </ForwardRef>,
      "href": Object {
        "pathname": "items/",
        "query": Object {
          "id": undefined,
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3
            className="title"
          >
            Boots
          </h3>,
          <div
            className="image-wrapper"
          >
            <img
              alt="Boots"
              className="img-fluid"
              src="/a.png"
            />
          </div>,
          <div
            className="right"
          >
            <small>
              category: 
              1
            </small>
            <p>
              Some boots
            </p>
            <div
              className="horizontal"
            >
              <button
                className="btn btn-outline-primary"
              >
                Contact Seller
              </button>
              <p
                className="price"
              >
                $12.00
              </p>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Boots",
            "className": "title",
          },
          "ref": null,
          "rendered": "Boots",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Boots"
              className="img-fluid"
              src="/a.png"
            />,
            "className": "image-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Boots",
              "className": "img-fluid",
              "src": "/a.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <small>
                category: 
                1
              </small>,
              <p>
                Some boots
              </p>,
              <div
                className="horizontal"
              >
                <button
                  className="btn btn-outline-primary"
                >
                  Contact Seller
                </button>
                <p
                  className="price"
                >
                  $12.00
                </p>
              </div>,
            ],
            "className": "right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "category: ",
                  1,
                ],
              },
              "ref": null,
              "rendered": Array [
                "category: ",
                1,
              ],
              "type": "small",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Some boots",
              },
              "ref": null,
              "rendered": "Some boots",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-outline-primary"
                  >
                    Contact Seller
                  </button>,
                  <p
                    className="price"
                  >
                    $12.00
                  </p>,
                ],
                "className": "horizontal",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Contact Seller",
                    "className": "btn btn-outline-primary",
                  },
                  "ref": null,
                  "rendered": "Contact Seller",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "$12.00",
                    "className": "price",
                  },
                  "ref": null,
                  "rendered": "$12.00",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "Item__ItemWrapper-hyead2-0",
          "isStatic": false,
          "rules": Array [
            "border:2px solid lightgray;box-shadow:",
            [Function],
            ";border-radius:0.5rem;padding:1rem;margin:1rem;display:grid;grid-template-areas:\\"title\\" \\"image\\" \\"right\\";text-align:center;justify-content:center;@media (min-width:600px){text-align:left;grid-template-columns:minmax(100px,400px) 1fr;grid-template-rows:auto 1fr;grid-template-areas:\\"image title\\" \\"image right\\";}.title{grid-area:title;}.right{display:flex;flex-direction:column;width:100%;grid-area:right;}.image-wrapper{margin:auto 0.5rem;flex:initial;grid-area:image;}.price{color:green;font-weight:600;padding-left:0.5rem;}.horizontal{width:100%;display:inline-flex;justify-content:space-between;align-items:center;margin-top:auto;p{margin:0;}}",
          ],
        },
        "displayName": "Item__ItemWrapper",
        "render": [Function],
        "styledComponentId": "Item__ItemWrapper-hyead2-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <h3
            className="title"
          >
            Boots
          </h3>
          <div
            className="image-wrapper"
          >
            <img
              alt="Boots"
              className="img-fluid"
              src="/a.png"
            />
          </div>
          <div
            className="right"
          >
            <small>
              category: 
              1
            </small>
            <p>
              Some boots
            </p>
            <div
              className="horizontal"
            >
              <button
                className="btn btn-outline-primary"
              >
                Contact Seller
              </button>
              <p
                className="price"
              >
                $12.00
              </p>
            </div>
          </div>
        </ForwardRef>,
        "href": Object {
          "pathname": "items/",
          "query": Object {
            "id": undefined,
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h3
              className="title"
            >
              Boots
            </h3>,
            <div
              className="image-wrapper"
            >
              <img
                alt="Boots"
                className="img-fluid"
                src="/a.png"
              />
            </div>,
            <div
              className="right"
            >
              <small>
                category: 
                1
              </small>
              <p>
                Some boots
              </p>
              <div
                className="horizontal"
              >
                <button
                  className="btn btn-outline-primary"
                >
                  Contact Seller
                </button>
                <p
                  className="price"
                >
                  $12.00
                </p>
              </div>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Boots",
              "className": "title",
            },
            "ref": null,
            "rendered": "Boots",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Boots"
                className="img-fluid"
                src="/a.png"
              />,
              "className": "image-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Boots",
                "className": "img-fluid",
                "src": "/a.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <small>
                  category: 
                  1
                </small>,
                <p>
                  Some boots
                </p>,
                <div
                  className="horizontal"
                >
                  <button
                    className="btn btn-outline-primary"
                  >
                    Contact Seller
                  </button>
                  <p
                    className="price"
                  >
                    $12.00
                  </p>
                </div>,
              ],
              "className": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "category: ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "category: ",
                  1,
                ],
                "type": "small",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Some boots",
                },
                "ref": null,
                "rendered": "Some boots",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="btn btn-outline-primary"
                    >
                      Contact Seller
                    </button>,
                    <p
                      className="price"
                    >
                      $12.00
                    </p>,
                  ],
                  "className": "horizontal",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Contact Seller",
                      "className": "btn btn-outline-primary",
                    },
                    "ref": null,
                    "rendered": "Contact Seller",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "$12.00",
                      "className": "price",
                    },
                    "ref": null,
                    "rendered": "$12.00",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Item__ItemWrapper-hyead2-0",
            "isStatic": false,
            "rules": Array [
              "border:2px solid lightgray;box-shadow:",
              [Function],
              ";border-radius:0.5rem;padding:1rem;margin:1rem;display:grid;grid-template-areas:\\"title\\" \\"image\\" \\"right\\";text-align:center;justify-content:center;@media (min-width:600px){text-align:left;grid-template-columns:minmax(100px,400px) 1fr;grid-template-rows:auto 1fr;grid-template-areas:\\"image title\\" \\"image right\\";}.title{grid-area:title;}.right{display:flex;flex-direction:column;width:100%;grid-area:right;}.image-wrapper{margin:auto 0.5rem;flex:initial;grid-area:image;}.price{color:green;font-weight:600;padding-left:0.5rem;}.horizontal{width:100%;display:inline-flex;justify-content:space-between;align-items:center;margin-top:auto;p{margin:0;}}",
            ],
          },
          "displayName": "Item__ItemWrapper",
          "render": [Function],
          "styledComponentId": "Item__ItemWrapper-hyead2-0",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
